src/pudl/helpers.py

```diff
@@ -1918,3 +1923,84 @@ def diff_wide_tables(
     return TableDiff(
         deleted=deleted, added=added, changed=changed, old_df=old, new_df=new
     )
+
+
+def create_datasette_metadata_yaml() -> str:
+    """Create datasette metadata yaml.
+
+    Returns:
+        Datasette metadata for all PUDL resources as YAML
+    """
+    pudl_output = PudlPaths().pudl_output
+    metadata = DatasetteMetadata.from_data_source_ids(pudl_output)
+    return metadata.to_yaml()
+
+
+def parse_datasette_metadata_yml(metadata_yml: str) -> dict:
+    """Parse a yaml file of datasette metadata as json.
+
+    Args:
+        metadata_yml: datasette metadata as yml.
+
+    Returns:
+        Parsed datasette metadata as JSON.
+    """
+    metadata_json = json.dumps(yaml.safe_load(metadata_yml))
+    return datasette.utils.parse_metadata(metadata_json)
+
+
+def check_tables_have_metadata(
```

Comment:
This check gets run when we deploy to datasette using `devtools/datasette/publish.py` and after the ETL is run in the integration tests. This way we'll catch errors in the CI and during datasette deployments.